#http://retractiondatabase.org/RetractionSearch.aspx?

Anlyse infomrmation about papers and retractions directly taken from this webpage. 
Type: article or letter.
Reasons: Error in data, error in results and/or conclusions


In [1]:
import os
os.getcwd()
path = "/project/eLife_retractions"
os.chdir(path)

In [2]:
import pandas as pd

df =pd.read_csv('retractions600.txt', sep = '\t')

df = df.reset_index()
df

level_0  \
0          600 Items Displayed Out of 791 Item(s) Found   
1     Structure revision of aspergicin by the crysta...   
2     (PHY) Chemistry; (PHY) Crystallography/Spectro...   
3        Natural Product Research ---Taylor and Francis   
4     Department of Chemical Engineering, Foshan Uni...   
5     http://retractionwatch.com/2018/01/16/caught-n...   
6                                                   NaN   
7                                                   NaN   
8     Solar photothermochemical alkane reverse combu...   
9                        (PHY) Chemistry; (PHY) Energy;   
10    Proceedings of the National Academy of Science...   
11    Department of Chemistry and Biochemistry, Univ...   
12    Car safety seats for children: rear facing for...   
13    (B/T) Transportation; (HSC) Medicine - Pediatr...   
14                  Injury Prevention ---BMJ Publishing   
15    Center for Applied Biomechanics, University of...   
16    http://retractionwatch.com/2017/12/15/saving-k...   
17                                                  NaN   
18                                                  NaN   
19    Car safety seats for children: rear facing for...   
20    (B/T) Transportation; (HSC) Medicine - Pediatr...   
21                  Injury Prevention ---BMJ Publishing   
22    Center for Applied Biomechanics, University of...   
23    http://retractionwatch.com/2017/12/15/saving-k...   
24                                                  NaN   
25                                                  NaN   
26    Transgenic mice overexpressing the ALS-linked ...   
27    (BLS) Biology - Cellular; (BLS) Biology - Mole...   
28    Acta Neuropathologica Communications ---Spring...   
29    Center for Translational Research in Neurodege...   
...                                                 ...   
4032                                                NaN   
4033                                                NaN   
4034                                                NaN   
4035                                                NaN   
4036                                                NaN   
4037  Expression of matrix metalloproteinase-26 (MMP...   
4038  (BLS) Anatomy/Physiology; (BLS) Biology - Cell...   
4039                          Life Sciences ---Elsevier   
4040  State Key Laboratory of Reproductive Biology, ...   
4041                                                NaN   
4042                                                NaN   
4043                                                NaN   
4044                                                NaN   
4045  [NEt4][AuCl2][AuCl3]: Solid-State Evidence of ...   
4046  (PHY) Chemistry; (PHY) Crystallography/Spectro...   
4047                         Angewandte Chemie ---Wiley   
4048  School of Chemistry, University of Manchester,...   
4049  Origin of the Obliquities of the Giant Planets...   
4050  (PHY) Space - Astronomy; (PHY) Space - Astroph...   
4051                  Nature ---Nature Publishing Group   
4052  Facultad de Ciencias Astron�micas y Geof�sicas...   
4053  Diffractive electron imaging of nanoparticles ...   
4054          (PHY) Chemistry; (PHY) Materials Science;   
4055  Nature Materials ---Springer - Nature Publishi...   
4056  Department of Physics and Astronomy, Arizona S...   
4057  YRKL sequence of influenza virus M1 functions ...   
4058  (BLS) Biology - Cellular; (BLS) Genetics; (HSC...   
4059  Journal of Virology ---American Society for Mi...   
4060  Department of Microbiology, Immunology and Mol...   
4061                                                NaN   

                                    level_1                 level_2  \
0                                       NaN                     NaN   
1      +Error in Results and/or Conclusions                Feng Zhu   
2                                       NaN              Jie-sen Li   
3                                       NaN             Wu-chen Xie   
4                                 

In [3]:
df2 = df[['level_0','Date/PubMedID/DOI', 'Date/PubMedID/DOI.1']]
df2 = df2.dropna(subset = ['Date/PubMedID/DOI'])
df2 = df2.dropna().reset_index(drop = 'True')
df2

level_0  \
0     Structure revision of aspergicin by the crysta...   
1     (PHY) Chemistry; (PHY) Crystallography/Spectro...   
2        Natural Product Research ---Taylor and Francis   
3     Solar photothermochemical alkane reverse combu...   
4                        (PHY) Chemistry; (PHY) Energy;   
5     Proceedings of the National Academy of Science...   
6     Car safety seats for children: rear facing for...   
7     (B/T) Transportation; (HSC) Medicine - Pediatr...   
8                   Injury Prevention ---BMJ Publishing   
9     Car safety seats for children: rear facing for...   
10    (B/T) Transportation; (HSC) Medicine - Pediatr...   
11                  Injury Prevention ---BMJ Publishing   
12    Transgenic mice overexpressing the ALS-linked ...   
13    (BLS) Biology - Cellular; (BLS) Biology - Mole...   
14    Acta Neuropathologica Communications ---Spring...   
15    Schisandrin B suppresses NLRP3 inflammasome ac...   
16    (BLS) Biochemistry; (BLS) Biology - Cellular; ...   
17    Biochemical and Biophysical Research Communica...   
18    Transition-Metal-Free C-CN/C-H Cross-Coupling:...   
19                                     (PHY) Chemistry;   
20    Organic Letters ---American Chemical Society (...   
21    NAD(P)H:quinone oxidoreductase 1 protects lung...   
22    (BLS) Biochemistry; (BLS) Biology - Cellular; ...   
23          Free Radical Biology & Medicine ---Elsevier   
24    Effects of synthetic magnetic flux in a parity...   
25                                       (PHY) Physics;   
26       Physical Review A ---American Physical Society   
27    Whole-Genome De Novo Sequencing of the Lignin-...   
28    (BLS) Genetics; (BLS) Microbiology; (BLS) Zool...   
29    Genome Announcements ---American Society for M...   
...                                                 ...   
1608  Structure of MsbA from E. coli: a homolog of t...   
1609  (BLS) Biochemistry; (BLS) Biology - Molecular;...   
1610  Science ---American Association for the Advanc...   
1611  Structure of the ABC transporter MsbA in compl...   
1612  (BLS) Biochemistry; (BLS) Biology - Molecular;...   
1613  Science ---American Association for the Advanc...   
1614  X-ray structure of the EmrE multidrug transpor...   
1615  (BLS) Biology - Molecular; (BLS) Microbiology;...   
1616  Science ---American Association for the Advanc...   
1617  Dual signaling regulated by calcyon, a D1 dopa...   
1618  (BLS) Biochemistry; (BLS) Biology - Molecular;...   
1619  Science ---American Association for the Advanc...   
1620  Silencing of B cell receptor signals in human ...   
1621                       (HSC) Medicine - Immunology;   
1622  The Journal of Experimental Medicine ---Rockef...   
1623  Expression of matrix metalloproteinase-26 (MMP...   
1624  (BLS) Anatomy/Physiology; (BLS) Biology - Cell...   
1625                          Life Sciences ---Elsevier   
1626  [NEt4][AuCl2][AuCl3]: Solid-State Evidence of ...   
1627  (PHY) Chemistry; (PHY) Crystallography/Spectro...   
1628                         Angewandte Chemie ---Wiley   
1629  Origin of the Obliquities of the Giant Planets...   
1630  (PHY) Space - Astronomy; (PHY) Space - Astroph...   
1631                  Nature ---Nature Publishing Group   
1632  Diffractive electron imaging of nanoparticles ...   
1633          (PHY) Chemistry; (PHY) Materials Science;   
1634  Nature Materials ---Springer - Nature Publishi...   
1635  YRKL sequence of influenza virus M1 functions ...   
1636  (BLS) Biology - Cellular; (BLS) Genetics; (HSC...   
1637  Journal of Virology ---American Society for Mi...   

                        Date/PubMedID/DOI                  Date/PubMedID/DOI.1  
0                               3/13/2017                             1/8/2018  
1                                28287288                             29310452  
2           10.1080/14786419.2017.1300800        10.1080/14786419.2017.1422331  
3                                3/8/2016                        

In [4]:
df2.shape[0] % 3

0

In [5]:
from datetime import datetime
from dateutil.parser import parse
import pandas as pd

DOI_article = []
PubMed_ID_article = []
DOI_retraction = []
PubMed_ID_retraction = []
#Journal = []
date_article = []
date_retraction = []

for index, row in df2.iterrows():
    if len(row['Date/PubMedID/DOI']) > 10: 
        try:
            date_article.append(datetime.strptime(df2.iloc[index -2, 1], '%m/%d/%Y'))
            date_retraction.append(datetime.strptime(df2.iloc[index -2, 2], '%m/%d/%Y'))

            DOI_article.append(row['Date/PubMedID/DOI'])
            DOI_retraction.append(row['Date/PubMedID/DOI.1'])
        
            PubMed_ID_article.append(df2.iloc[index -1, 1])
            PubMed_ID_retraction.append(df2.iloc[index -1, 2])
            
        except ValueError:
            pass
                      
        
        #print(row['Date/PubMedID/DOI'],row['Date/PubMedID/DOI.1'])

# df2 = pd.DataFrame({'Journal': Journal,'DOI':DOI_a
df_retracted_retraction = pd.DataFrame({'doi':DOI_article, 
                    'pmid_article':PubMed_ID_article, 
                    'doi_retraction':DOI_retraction, 
                    'pmid_retraction':PubMed_ID_retraction,
                    'date_article': date_article,
                    'date_retraction' : date_retraction}).drop_duplicates()

#mask = df_retracted_retraction.date_retraction != 'unavailable'
#mask4 = df_retracted_retraction.date_retraction != "0"
#df_retracted_retraction = df_retracted_retraction[mask3 & mask4]
df_retracted_retraction



date_article date_retraction                                  doi  \
0     2017-03-13      2018-01-08        10.1080/14786419.2017.1300800   
1     2016-03-08      2018-01-03              10.1073/pnas.1516945113   
2     2007-12-01      2017-12-20               10.1136/ip.2006.015115   
3     2007-12-01      2017-06-30               10.1136/ip.2006.015115   
4     2016-11-18      2017-12-13            10.1186/s40478-016-0393-5   
5     2017-08-31      2017-12-09           10.1016/j.bbrc.2017.08.079   
6     2017-09-11      2017-11-30          10.1021/acs.orglett.7b02473   
7     2012-02-01      2017-11-20  10.1016/j.freeradbiomed.2011.11.027   
8     2017-10-10      2017-11-16           10.1103/PhysRevA.96.043821   
9     2017-08-10      2017-11-16             10.1128/genomeA.00731-17   
10    2017-05-03      2017-11-13                   10.1039/C7TA02103H   
11    2014-12-19      2017-10-26        10.1126/science.346.6216.1486   
12    2017-10-01      2017-10-25            10.21873/anticanres.11969   
13    2012-10-01      2017-10-20      10.1001/archpediatrics.2012.999   
14    2012-10-01      2017-09-21      10.1001/archpediatrics.2012.999   
15    2008-01-01      2017-10-14                   10.1093/ndt/gfm690   
16    2016-09-20      2017-10-03              10.1021/acs.joc.6b02086   
17    2016-05-11      2017-10-02        10.1158/1535-7163.MCT-15-0375   
18    2016-09-09      2017-10-01                          unavailable   
19    2017-07-25      2017-09-28                10.3892/mmr.2017.7084   
20    2014-12-02      2017-09-26                 10.3892/or.2014.3644   
21    2014-12-26      2017-09-26         10.3109/19401736.2014.987245   
22    2017-04-18      2017-09-19               10.1001/jama.2017.1469   
23    2017-05-11      2017-09-15            10.1186/s13033-017-0143-5   
24    2017-02-23      2017-09-14         10.1021/acs.jmedchem.6b01417   
25    2017-01-25      2017-09-08                    10.7150/jca.15753   
26    2017-07-03      2017-09-06                   10.1039/c7ew00132k   
27    2017-01-31      2017-09-05               10.1038/sdata.2016.120   
28    2017-07-05      2017-09-05                10.1139/cjp-2017-0290   
29    2017-07-08      2017-09-05           10.1016/j.gene.2017.07.018   
..           ...             ...                                  ...   
421   2005-10-03      2008-04-17            10.1016/j.jep.2005.08.060   
422   2005-04-08      2008-04-17            10.1016/j.jep.2005.01.027   
423   2004-12-17      2008-04-11              10.1126/science.1103776   
424   2001-11-08      2008-03-06                     10.1038/35102506   
425   2004-07-06      2008-02-20              10.1073/pnas.0403155101   
426   2006-02-08      2008-02-14            10.1016/j.jmb.2006.01.087   
427   2006-07-06      2008-01-31                  10.1038/nature04836   
428   2007-12-11      2008-01-05              10.2310/6650.2007.00027   
429   2001-07-01      2007-12-15              10.1182/blood.V98.1.125   
430   2007-06-01      2007-12-01           10.1152/ajpendo.00702.2006   
431   2004-08-05      2007-10-31             10.1038/sj.emboj.7600343   
432   1998-02-06      2007-10-11                    10.1021/bi972113+   
433   2004-02-17      2007-10-09            10.1016/j.cub.2004.02.003   
434   2007-08-02      2007-10-01           10.1016/j.epsl.2007.07.046   
435   2002-06-21      2007-08-10              10.1126/science.1071776   
436   2007-03-16      2007-07-20         10.1371/journal.pcbi.0030051   
437   2006-07-01      2007-07-01           10.1016/j.aaen.2006.05.005   
438   2006-08-01      2007-06-13                    10.1021/bi060547b   
439   2005-06-10      2007-06-13                    10.1021/bi050786j   
440   2004-06-01      2007-02-01                10.1104/pp.103.037770   
441   2001-09-07      2006-12-22        10.1126/science.293.5536.1793   
442   2005-05-13      2006-12-22              10.1126/science.1107733   
443   2005-12-23      2006-12-22              10.1126/science.

# Let's use citation_context to get all the manuscripts citing these dois from PMC OAS

In [6]:
df_retracted_retraction.doi[445:]

445               10.1084/jem.20020881
446          10.1016/j.lfs.2005.05.045
447             10.1002/anie.200602092
448                10.1038/nature04577
449                   10.1038/nmat1531
450    10.1128/JVI.80.5.2291-2308.2006
Name: doi, dtype: object

In [ ]:
# using get_papers notebook and module/bs_preprocess.py from citation_context

DOI_list = df_retracted_retraction.doi[400:]
ds_name = 'df_retracted_retraction_400_450'


%store DOI_list
%store ds_name

%run get_papers.ipynb

Stored 'DOI_list' (Series)
Stored 'ds_name' (str)
10.1128/IAI.69.8.4858-4869.2001
data/FTP_PUBMED_papers/comm_use.A-B.xml.tar.gz
data/FTP_PUBMED_papers/comm_use.C-H.xml.tar.gz
data/FTP_PUBMED_papers/comm_use.I-N.xml.tar.gz
data/FTP_PUBMED_papers/comm_use.O-Z.xml.tar.gz


#### Let's read the files with the xml text.

In [7]:
import os
import pandas as pd
import glob 



data_path = 'data'
analysis_path = os.path.join(data_path, 'analysis')

df_articles_citing_retracted_paper = pd.DataFrame()


for filename in sorted(glob.glob(analysis_path + '/df_*0.tsv')):
    print(filename)

    df_in_file = pd.read_csv(filename, sep='\t', encoding='utf-8')
    print(df_in_file.index[-1])
    
    
    df_articles_citing_retracted_paper = df_articles_citing_retracted_paper.append(df_in_file)
    
df_articles_citing_retracted_paper.reset_index(drop = True)


data/analysis/df_retracted_retraction_0_100.tsv
68
data/analysis/df_retracted_retraction_100_200.tsv
93
data/analysis/df_retracted_retraction_200_300.tsv
84
data/analysis/df_retracted_retraction_300_400.tsv
127
data/analysis/df_retracted_retraction_400_450.tsv
28


DOI_cited  \
0                 10.1136/ip.2006.015115   
1                 10.1136/ip.2006.015115   
2              10.1186/s40478-016-0393-5   
3    10.1016/j.freeradbiomed.2011.11.027   
4               10.1128/genomeA.00731-17   
5          10.1126/science.346.6216.1486   
6                  10.3892/mmr.2017.7084   
7              10.1186/s13033-017-0143-5   
8              10.1186/s13033-017-0143-5   
9                      10.7150/jca.15753   
10                10.1038/sdata.2016.120   
11                    10.1038/ncomms6446   
12                    10.1038/ncomms6446   
13                    10.1093/nar/gks812   
14                 10.3892/mmr.2016.5294   
15                    10.1002/cbin.10615   
16                    10.1002/cbin.10291   
17                    10.1002/cbin.10291   
18          10.1371/journal.pone.0164378   
19                  10.1021/jacs.6b01652   
20          10.1371/journal.pone.0164571   
21          10.1371/journal.pone.0164571   
22               10.1186/1472-6882-14-82   
23               10.1186/1472-6882-14-82   
24                  10.1155/2017/6258146   
25                 10.1104/pp.110.157180   
26                   10.1289/ehp.1408282   
27                   10.1289/ehp.1408282   
28             10.1186/s12889-017-4082-5   
29          10.1371/journal.pone.0121549   
..                                   ...   
375                10.1096/fj.06-6171fje   
376                    10.1104/pp.011019   
377                    10.1104/pp.011019   
378              10.1073/pnas.0501997102   
379              10.1073/pnas.0501997102   
380                        10.1038/77997   
381   10.1213/01.ane.0000221167.58135.3d   
382   10.1213/01.ane.0000221167.58135.3d   
383   10.1213/01.ane.0000221167.58135.3d   
384   10.1213/01.ane.0000221167.58135.3d   
385   10.1213/01.ane.0000221167.58135.3d   
386                     10.1038/35044106   
387            10.4049/jimmunol.170.1.64   
388                10.1055/s-2006-924577   
389                10.1055/s-2006-924577   
390                10.1055/s-2006-924577   
391           10.1016/j.cell.2008.01.051   
392           10.1016/j.cell.2008.01.051   
393     10.1111/j.1365-2958.2005.04612.x   
394            10.1016/j.str.2008.04.010   
395            10.1016/j.str.2008.04.010   
396                    10.1002/ajh.20284   
397                     10.1038/35102506   
398                     10.1038/35102506   
399                    10.1021/bi050786j   
400                10.1104/pp.103.037770   
401              10.1126/science.1107733   
402              10.1126/science.1119776   
403                 10.1084/jem.20020881   
404                 10.1084/jem.20020881   

                                        article_citing  
0    <html>\n <body>\n  <pub-id pub-id-type="pmid">...  
1    <html>\n <body>\n  <pub-id pub-id-type="pmid">...  
2    <html>\n <body>\n  <article article-type="rese...  
3    <html>\n <body>\n  <comment>\n   S0891-\n  </c...  
4    <html>\n <body>\n  <article article-type="brie...  
5    <html>\n <body>\n  <pub-id pub-id-type="doi">\...  
6    <html>\n <body>\n  <article article-type="rese...  
7    <html>\n <body>\n  <article article-type="rese...  
8    <html>\n <body>\n  <p>\n   Int_J_Ment_Health_S...  
9    <html>\n <body>\n  <article article-type="rese...  
10   <html>\n <body>\n  <article article-type="data...  
11   <html>\n <body>\n  <article article-type="retr...  
12   <html>\n <body>\n  <p>\n   Sci_Rep/PMC5524909....  
13   <html>\n <body>\n  <pub-id pub-id-type="doi">\...  
14   <html>\n <body>\n  <article article-type="retr...  
15   <html>\n <body>\n  <article-title>\n   Mesench...  
16   <html>\n <body>\n  <article article-type="rese...  
17   <html>\n <body>\n  <p>\n   Biosci_Rep/PMC50258...  
18   <html>\n <body>\n  <article article-type="rese...  
19   <html>\n <body>\n  <year>\n   2016\n  </year>\...  
20   <html>\n <body>\n  <article article-type="rese...  
21   \n   <floats-group>\n    <fig id="ijerph-14

# REMOVE THE TEXT OF THE ORIGINAL PAPER 
# YOU MAY AVOID THIS BY INTRODUCING A GAP BERORE and AFTE RETRACTION

In [8]:
from modules import Article_object

list_Articles_citing = []
list_Article_citing_dois = []
list_Article_citing_dates = []
list_Article_citing_references = []


texto = df_articles_citing_retracted_paper['article_citing'].iloc[4]

Article_citing = Article_object.Article_object()
Article_citing.text = texto

Article_citing.get_doi_from_text()
Article_citing.get_date_from_text()
    
Article_citing.get_list_references_dois_and_pmids_from_text()
Article_citing.get_pmid_from_text()
Article_citing.describe()

PMID not found
pmc: 	  
pmid: 	  
doi: 	 10.1128/genomeA.00731-17 
journal: 	  
date: 	 2017-08-10 
text: 	 <html>
 <body>
  <article article-type="brief-report" xmlns:mml="http://www.w3.org/1998/Math/MathML" 
references_dois: 	 


In [9]:
from modules import Article_object

list_Articles_citing = []
list_Article_citing_dois = []
list_Article_citing_pmids = []
list_Article_citing_dates = []
list_Article_citing_references = []


for index, row in df_articles_citing_retracted_paper.iterrows():
    print(index)
    
    #for text in row['article_citing']:
    
    Article_citing = Article_object.Article_object()
    Article_citing.text = row['article_citing']

    Article_citing.get_doi_from_text()
    list_Article_citing_dois.append(Article_citing.doi)

    Article_citing.get_pmid_from_text()
    list_Article_citing_pmids.append(Article_citing.pmid)
    
    Article_citing.get_date_from_text()
    list_Article_citing_dates.append(Article_citing.date)

    Article_citing.get_list_references_dois_and_pmids_from_text()
    list_Article_citing_references.append(Article_citing.references_dois)
    
    

    list_Articles_citing.append(Article_citing)

    #df_Article_ref_retraction = {"doi": row["doi"],
    #                             "pmid":row["pmid"],
    #                             "citing_article": article_ref_retraction,
    #                             "critizising": 1}

    #list_Articles_criticising.append(df_Article_ref_retraction)

"""
df_Articles_citing_retracted_paper = pd.concat([df_articles_citing_retracted_paper, 
                                    pd.DataFrame({"Article": list_Articles_citing,
                                            "date_article_citing": list_Article_citing_dates,
                                            "doi_article_citing": list_Article_citing_dois,
                                            "pmid_article_citing": list_Article_citing_pmids,
                                            "references_in_article_citing": list_Article_citing_references                     
                                            })
                                    ], axis=1)
"""


0
1
2
PMID not found
3
DOI not found
Date not found
4
PMID not found
5
DOI not found
Date not found
6
7
8
DOI not found
9
PMID not found
10
PMID not found
11
12
PMID not found
13
Day or month out of range!!!
day:  
         31
         month:  
         11
         year:  
      2010
     
Date set to: 
day:  29 month:  
         11
         year:  
      2010
     
14
PMID not found
15
DOI not found
Date not found
16
17
DOI not found
Date not found
18
PMID not found
19
DOI not found
Date not found
20
21
PMID not found
22
23
DOI not found
Date not found
24
PMID not found
25
DOI not found
Date not found
26
27
28
29
DOI not found
Date not found
30
DOI not found
Date not found
31
32
PMID not found
33
DOI not found
Date not found
34
DOI not found
Date not found
35
DOI not found
Date not found
36
DOI not found
Date not found
37
DOI not found
Date not found
38
DOI not found
Date not found
39
DOI not found
Date not found
40
DOI not found
Date not found
41
DOI not found
Date not found
42
DOI n

'\ndf_Articles_citing_retracted_paper = pd.concat([df_articles_citing_retracted_paper, \n                                    pd.DataFrame({"Article": list_Articles_citing,\n                                            "date_article_citing": list_Article_citing_dates,\n                                            "doi_article_citing": list_Article_citing_dois,\n                                            "pmid_article_citing": list_Article_citing_pmids,\n                                            "references_in_article_citing": list_Article_citing_references                     \n                                            })\n                                    ], axis=1)\n'

In [10]:
string = "\n 25662882\n"
string = string.strip()
string

'25662882'

In [11]:
df_Articles_citing_retracted_paper = pd.concat([df_articles_citing_retracted_paper.reset_index(drop=True), 
                                    pd.DataFrame({"Article": list_Articles_citing,
                                            "date_article_citing": list_Article_citing_dates,
                                            "doi_article_citing": list_Article_citing_dois,
                                            "pmid_article_citing": list_Article_citing_pmids,
                                            "pmid_article_citing": list_Article_citing_pmids,
                                            "references_in_article_citing": list_Article_citing_references                     
                                            })
                                    ], axis=1)
df_Articles_citing_retracted_paper

# Let's add the date of the retraction
list_date_retracion = []
for  doi in df_Articles_citing_retracted_paper["DOI_cited"]:
    mask = df_retracted_retraction["doi"] == doi
    list_date_retracion.append(datetime.date(df_retracted_retraction[mask]["date_retraction"].iloc[0]))

df_Articles_citing =  pd.concat([df_Articles_citing_retracted_paper,
                                 pd.DataFrame({"date_doi_cited_retraction": list_date_retracion
                                               })
                                ], axis = 1)

In [12]:
df_Articles_citing.shape

(405, 8)

In [13]:
list_date_retracion[:5]

[datetime.date(2017, 12, 20),
 datetime.date(2017, 12, 20),
 datetime.date(2017, 12, 13),
 datetime.date(2017, 11, 20),
 datetime.date(2017, 11, 16)]

In [14]:
df_Articles_citing.head(5)

DOI_cited  \
0               10.1136/ip.2006.015115   
1               10.1136/ip.2006.015115   
2            10.1186/s40478-016-0393-5   
3  10.1016/j.freeradbiomed.2011.11.027   
4             10.1128/genomeA.00731-17   

                                      article_citing  \
0  <html>\n <body>\n  <pub-id pub-id-type="pmid">...   
1  <html>\n <body>\n  <pub-id pub-id-type="pmid">...   
2  <html>\n <body>\n  <article article-type="rese...   
3  <html>\n <body>\n  <comment>\n   S0891-\n  </c...   
4  <html>\n <body>\n  <article article-type="brie...   

                                             Article date_article_citing  \
0  <modules.Article_object.Article_object object ...          1993-08-30   
1  <modules.Article_object.Article_object object ...          1993-08-30   
2  <modules.Article_object.Article_object object ...          2016-11-18   
3  <modules.Article_object.Article_object object ...                       
4  <modules.Article_object.Article_object object ...          2017-08-10   

           doi_article_citing pmid_article_citing  \
0  10.1136/ip.2006.015115eoc1            25662882   
1  10.1136/ip.2006.015115eoc1            25662882   
2   10.1186/s40478-016-0393-5                       
3                                        22198263   
4    10.1128/genomeA.00731-17                       

  references_in_article_citing date_doi_cited_retraction  
0                                             2017-12-20  
1                                             2017-12-20  
2                                             2017-12-13  
3                                             2017-11-20  
4                                             2017-11-16

In [15]:
df_Articles_citing["Article"].iloc[401].text

'<html>\n <body>\n  <article-title>\n   Structure of P-glycoprotein reveals a molecular basis for poly-specific drug binding\n  </article-title>\n  .\n  <source>\n   Science\n  </source>\n  .\n  <year>\n   2009\n  </year>\n  ;\n  <volume>\n   323\n  </volume>\n  (\n  <issue>\n   5922\n  </issue>\n  ):\n  <fpage>\n   1718\n  </fpage>\n  –\n  <lpage>\n   22\n  </lpage>\n  .\n  <comment>\n   doi:\n   <ext-link ext-link-type="uri" xlink:href="http://dx.doi.org/10.1126/science.1168750">\n    10.1126/science.1168750\n   </ext-link>\n  </comment>\n  ;\n  <pub-id pub-id-type="pmid">\n   19325113\n  </pub-id>\n  <ref id="pone.0178238.ref020">\n   <label>\n    20\n   </label>\n   <mixed-citation publication-type="journal">\n    <name>\n     <surname>\n      Jin\n     </surname>\n     <given-names>\n      MS\n     </given-names>\n    </name>\n    ,\n    <name>\n     <surname>\n      Oldham\n     </surname>\n     <given-names>\n      ML\n     </given-names>\n    </name>\n    ,\n    <name>\n     <s

In [16]:
import numpy as np
df_Articles_citing.replace('', np.nan, inplace=True)
df_Articles_citing2 = df_Articles_citing.dropna(subset = ["date_article_citing"])
df_Articles_citing_clean = df_Articles_citing2.reset_index(drop = True)
df_Articles_citing_clean

DOI_cited  \
0                10.1136/ip.2006.015115   
1                10.1136/ip.2006.015115   
2             10.1186/s40478-016-0393-5   
3              10.1128/genomeA.00731-17   
4                 10.3892/mmr.2017.7084   
5             10.1186/s13033-017-0143-5   
6             10.1186/s13033-017-0143-5   
7                     10.7150/jca.15753   
8                10.1038/sdata.2016.120   
9                    10.1038/ncomms6446   
10                   10.1038/ncomms6446   
11                   10.1093/nar/gks812   
12                10.3892/mmr.2016.5294   
13                   10.1002/cbin.10291   
14         10.1371/journal.pone.0164378   
15         10.1371/journal.pone.0164571   
16         10.1371/journal.pone.0164571   
17              10.1186/1472-6882-14-82   
18                 10.1155/2017/6258146   
19                  10.1289/ehp.1408282   
20                  10.1289/ehp.1408282   
21            10.1186/s12889-017-4082-5   
22         10.1371/journal.pone.0121549   
23         10.1371/journal.pone.0146809   
24           10.1016/j.cell.2013.04.008   
25           10.1016/j.cell.2013.04.008   
26           10.1016/j.cell.2013.04.008   
27           10.1016/j.cell.2013.04.008   
28           10.1016/j.cell.2013.04.008   
29           10.1016/j.cell.2013.04.008   
..                                  ...   
188                      10.1038/nm1491   
189                      10.1038/nm1491   
190                      10.1038/nm1491   
191                      10.1038/nm1491   
192            10.1200/JCO.2007.11.0593   
193             10.1073/pnas.0809516106   
194       10.1158/0008-5472.CAN-07-5937   
195             10.1126/science.1124755   
196             10.1126/science.1124755   
197             10.1126/science.1124755   
198       10.1158/0008-5472.CAN-04-4194   
199             10.1073/pnas.0503877102   
200               10.1083/jcb.200705182   
201               10.1083/jcb.200705182   
202               10.1083/jcb.200705182   
203                   10.1104/pp.011019   
204             10.1073/pnas.0501997102   
205             10.1073/pnas.0501997102   
206                       10.1038/77997   
207  10.1213/01.ane.0000221167.58135.3d   
208  10.1213/01.ane.0000221167.58135.3d   
209               10.1055/s-2006-924577   
210               10.1055/s-2006-924577   
211          10.1016/j.cell.2008.01.051   
212           10.1016/j.str.2008.04.010   
213                   10.1002/ajh.20284   
214                    10.1038/35102506   
215             10.1126/science.1119776   
216                10.1084/jem.20020881   
217                10.1084/jem.20020881   

                                        article_citing  \
0    <html>\n <body>\n  <pub-id pub-id-type="pmid">...   
1    <html>\n <body>\n  <pub-id pub-id-type="pmid">...   
2    <html>\n <body>\n  <article article-type="rese...   
3    <html>\n <body>\n  <article article-type="brie...   
4    <html>\n <body>\n  <article article-type="rese...   
5    <html>\n <body>\n  <article article-type="rese...   
6    <html>\n <body>\n  <p>\n   Int_J_Ment_Health_S...   
7    <html>\n <body>\n  <article article-type="rese...   
8    <html>\n <body>\n  <article article-type="data...   
9    <html>\n <body>\n  <article article-type="retr...   
10   <html>\n <body>\n  <p>\n   Sci_Rep/PMC5524909....   
11   <html>\n <body>\n  <pub-id pub-id-type="doi">\...   
12   <html>\n <body>\n  <article article-type="retr...   
13   <html>\n <body>\n  <article article-type="rese...   
14   <html>\n <body>\n  <article article-type="rese...   
15   <html>\n <body>\n  <article article-type="rese...   
16   \n   <floats-group>\n    <fig id="ijerph-14-00...   
17   <html>\n <body>\n  <article article-type="rese...   
18   <html>\n <body>\n  <article article-type="retr...   
19   <html>\n <body>\n  <article article-type="rese...   
20   <html>\n <body>\n  <p>\n   Environ_Health_Pers...   
21   <html>\n <body>\n  <article article-type="rese...   
22   \n <p>\n  Biol_Res/PMC44

In [17]:
df_Articles_citing_clean.shape

(218, 8)

list_index_is_the_retracted_paper = []

for index, row in df_Articles_citing.iterrows():   
    if row.DOI_cited == row.doi_article_citing:
        list_index_is_the_retracted_paper.append(index)
        #print(row.DOI_cited)
        #print(row.doi_article_citing)
        #print(index)
    
        
df_Retractions_found = df_Articles_citing_retracted_paper_clean.loc[list_index_is_the_retracted_paper]


In [18]:
list_index_is_the_retracted_paper =[]
list_index_is_the_retraction_note = []
for index, row in df_Articles_citing_clean.iterrows():   
    if row.doi_article_citing in list(df_retracted_retraction["doi"]):
        list_index_is_the_retracted_paper.append(index)
        
    if row.doi_article_citing in list(df_retracted_retraction["doi_retraction"]):
        list_index_is_the_retraction_note.append(index)
        #print(row.DOI_cited)
        #print(row.doi_article_citing)
        #print(index)

df_Articles_citing_no_retractions =  df_Articles_citing_clean.drop(list_index_is_the_retracted_paper + 
                                                                   list_index_is_the_retraction_note)


In [19]:
df_Articles_citing_no_retractions.shape

(152, 8)

In [20]:
len(list_index_is_the_retracted_paper)

42

In [21]:
len(list_index_is_the_retraction_note)

25

In [22]:
df_Articles_citing_no_retractions.columns

Index(['DOI_cited', 'article_citing', 'Article', 'date_article_citing',
       'doi_article_citing', 'pmid_article_citing',
       'references_in_article_citing', 'date_doi_cited_retraction'],
      dtype='object')

In [23]:
list_before_retraction = []
list_after_retraction = []

for index, row in df_Articles_citing_no_retractions.iterrows():
    if row['date_article_citing'] < row['date_doi_cited_retraction']:
        list_before_retraction.append(1)
        list_after_retraction.append(0)
    else:
        list_before_retraction.append(0)
        list_after_retraction.append(1)


df_Articles_before_after_retraction = pd.concat([df_Articles_citing_no_retractions.reset_index(drop = True),
                                                pd.DataFrame({"before_retraction": list_before_retraction,
                                                              "after_retraction": list_after_retraction
                                                             })
                                                ], axis = 1)
        

df_Articles_before_after_retraction        


DOI_cited  \
0             10.1186/s13033-017-0143-5   
1                    10.1002/cbin.10291   
2          10.1371/journal.pone.0164571   
3            10.1016/j.cell.2013.04.008   
4            10.1016/j.cell.2013.04.008   
5            10.1016/j.cell.2013.04.008   
6            10.1016/j.cell.2013.04.008   
7            10.1016/j.cell.2013.04.008   
8            10.1016/j.cell.2013.04.008   
9                10.1074/jbc.M306295200   
10               10.1074/jbc.M100194200   
11               10.1074/jbc.M100194200   
12           10.1016/j.cell.2010.09.011   
13              10.1186/1742-2094-9-223   
14                    10.1038/srep19222   
15                    10.1038/srep19222   
16                 10.1128/AEM.02433-12   
17                 10.1128/AEM.02433-12   
18              10.1126/science.1237242   
19               10.1098/rspb.2009.1774   
20           10.1016/j.cell.2012.02.055   
21           10.1152/ajplung.00134.2011   
22           10.1016/j.cell.2009.03.047   
23           10.1016/j.cell.2009.03.047   
24           10.1016/j.cell.2009.03.047   
25           10.1016/j.cell.2009.03.047   
26           10.1016/j.cell.2009.03.047   
27           10.1016/j.cell.2009.03.047   
28           10.1016/j.cell.2009.03.047   
29           10.1016/j.cell.2009.03.047   
..                                  ...   
122                      10.1038/nm1491   
123                      10.1038/nm1491   
124                      10.1038/nm1491   
125                      10.1038/nm1491   
126                      10.1038/nm1491   
127                      10.1038/nm1491   
128                      10.1038/nm1491   
129                      10.1038/nm1491   
130            10.1200/JCO.2007.11.0593   
131             10.1073/pnas.0809516106   
132       10.1158/0008-5472.CAN-07-5937   
133             10.1126/science.1124755   
134             10.1126/science.1124755   
135             10.1126/science.1124755   
136       10.1158/0008-5472.CAN-04-4194   
137             10.1073/pnas.0503877102   
138               10.1083/jcb.200705182   
139                   10.1104/pp.011019   
140             10.1073/pnas.0501997102   
141             10.1073/pnas.0501997102   
142                       10.1038/77997   
143  10.1213/01.ane.0000221167.58135.3d   
144  10.1213/01.ane.0000221167.58135.3d   
145               10.1055/s-2006-924577   
146               10.1055/s-2006-924577   
147          10.1016/j.cell.2008.01.051   
148           10.1016/j.str.2008.04.010   
149                   10.1002/ajh.20284   
150                    10.1038/35102506   
151             10.1126/science.1119776   

                                        article_citing  \
0    <html>\n <body>\n  <p>\n   Int_J_Ment_Health_S...   
1    <html>\n <body>\n  <article article-type="rese...   
2    <html>\n <body>\n  <article article-type="rese...   
3    \n              <floats-group>\n              ...   
4    \n                     Lipids_Health_Dis/PMC47...   
5    \n                       Diabetologia/PMC38578...   
6    <html>\n <body>\n  <p>\n   Int_J_Mol_Med/PMC53...   
7    \n  Islets/PMC4589993.nxml                    ...   
8    <html>\n <body>\n  <p>\n   J_Clin_Med_Res/PMC5...   
9    \n <p>\n  Lipids_Health_Dis/PMC3544701.nxml   ...   
10   \n <fpage>\n  20436\n </fpage>\n <p>\n  –\n  <...   
11   <html>\n <body>\n  <p>\n   Mol_Cancer/PMC29462...   
12   <html>\n <body>\n  <p>\n   quantile normalizat...   
13   \n  <p>\n   Mol_Autism/PMC3922109.nxml        ...   
14   <html>\n <body>\n  <article article-type="revi...   
15   \n <p>\n  Chin_J_Cancer/PMC5251273.nxml       ...   
16   \n <p>\n  Appl_Microbiol_Biotechnol/PMC5071362...   
17   <html>\n <body>\n  <p>\n   Biotechnol_Biofuels...   
18   <html>\n <body>\n  <article article-type="edit...   
19   \n <p>\n  BMC_Evol_Biol/PMC4587847.nxml       ...   
20   \n <p>\n  BMC_Infect_Dis/PMC4234735.nxml      ...   
21   \n <p>\n  BMC_Pulm_Med/PMC3907130.nxml        ...   
22   <html>\n <body>\n  <arti

In [24]:
df_Articles_before_after_retraction[df_Articles_before_after_retraction["before_retraction"]==1].shape

(107, 10)

In [25]:
len(list_retraction)

NameError: name 'list_retraction' is not defined

In [31]:
len(df_Articles_before_after_retraction["DOI_cited"].unique())

61

# Summary

61 different DOIs of retracted papers cited by
218 articles.

42 retracted papers, 25 retraction notes, and 152 new manuscripts.

152 new manuscripts:
     45 were published after the retraction
    107 before the retraction.

# Let's analyse the text and find the sentence were the paper was cited

In [26]:
df_Articles_before_after_retraction.tail(1)

DOI_cited  \
151  10.1126/science.1119776   

                                        article_citing  \
151  <html>\n <body>\n  <article article-type="rese...   

                                               Article date_article_citing  \
151  <modules.Article_object.Article_object object ...          2010-02-18   

           doi_article_citing pmid_article_citing  \
151  10.1186/1471-2164-11-120             3138193   

     references_in_article_citing date_doi_cited_retraction  after_retraction  \
151                           NaN                2006-12-22                 1   

     before_retraction  
151                  0

In [27]:
import imp
import sys
sys.path.insert(0, "./modules")
from modules import get_df_analise_section
#imp.reload()

list_DOI_cited = df_Articles_before_after_retraction.DOI_cited.unique()

df = pd.DataFrame()
for index, row in df_Articles_before_after_retraction.iterrows():
    df_analysis_row = get_df_analise_section.get_df(row.DOI_cited, [row.article_citing])
    df = df.append(df_analysis_row)
    
"""    
for doi in list_DOI_cited[3:4]:
    mask = df_Articles_before_after_retraction.DOI_cited == doi
    mask2 = df_Articles_before_after_retraction.after_retraction == 1
    list_text_articles = list(df_Articles_before_after_retraction[(mask)].article_citing)
    df = get_df_analise_section.get_df(doi, list_text_articles)

    
"""    
    
"""



for index, DOI in enumerate(DOI_list): 

    df_articles_citingDOI = df_articles[df_articles['DOI_cited'] == DOI]['article_citing']
    list_articles_citingDOI = df_articles_citingDOI.values.tolist()

    df = get_df_analise_section.get_df(DOI, list_articles_citingDOI)
"""

"\n\n\n\nfor index, DOI in enumerate(DOI_list): \n\n    df_articles_citingDOI = df_articles[df_articles['DOI_cited'] == DOI]['article_citing']\n    list_articles_citingDOI = df_articles_citingDOI.values.tolist()\n\n    df = get_df_analise_section.get_df(DOI, list_articles_citingDOI)\n"

In [32]:
df = df.reset_index(drop = True)
df.tail()

cited_DOI  cited_in_conclusions  cited_in_discussion  \
147  10.1016/j.cell.2008.01.051                  True                False   
148   10.1016/j.str.2008.04.010                 False                False   
149           10.1002/ajh.20284                 False                False   
150            10.1038/35102506                 False                False   
151     10.1126/science.1119776                 False                 True   

     cited_in_introduction  cited_in_maintext  \
147                  False              False   
148                   True               True   
149                  False              False   
150                  False              False   
151                  False              False   

                                   citing_DOI  conclusions_found  \
147  \n      10.1007/s00418-008-0430-y\n                    True   
148     \n      10.1186/1478-811X-9-22\n                   False   
149  \n      10.18632/oncotarget.14295\n                   False   
150  \n      10.1186/1471-2202-8-S3-S1\n                   False   
151   \n      10.1186/1471-2164-11-120\n                    True   

     discussion_found  introduction_found  maintext_found reference_number  \
147             False                True            True             CR48   
148             False                True            True              B26   
149              True                True            True               R5   
150             False               False            True              B30   
151              True                True            True              B58   

                           sentence_citing_conclusions  \
147  [<xref ref-type="bibr" rid="CR48">\n         2...   
148                                               None   
149                                               None   
150                                               None   
151                                               None   

                            sentence_citing_discussion  \
147                                               None   
148                                               None   
149                                               None   
150                                               None   
151  [</p>\n<p>\n      LGI1 also encoded a homolog ...   

                                 sentence_citing_intro  \
147                                               None   
148  [The mode of interaction of Gβ with binding pa...   
149                                               None   
150                                               None   
151                                               None   

                              sentence_citing_maintext  
147                                               None  
148  [The mode of interaction of Gβ with binding pa...  
149                                               None  
150                                               None  
151                                               None

df.sentence_citing_discussion.iloc[0]

In [65]:
#for index, row in df.iterrows():
import re
import numpy as np
from modules import words_frec_analysis_get_sentence

"""
df[['sentence_citing_conclusions',
       'sentence_citing_discussion', 'sentence_citing_intro',
       'sentence_citing_maintext']]
"""

def remove_citations(text):
    number_citations = text.count("[")
    while number_citations > 0:
        init = text.find("[") 
        end = text.find("]") 
        text = text.replace(text[init : end +1 ], " ") # replace the string between two delimiters
        number_citations -= 1
    return text

def remove_html_tags(text):
    to_clean = re.compile('<.*?>')
    clean_text = re.sub(to_clean, '', text)
    return clean_text
    
    
list_all_citing_citing_sentences = []

for index, row in df.iterrows():
    list_sentences_citing_conclusions = row['sentence_citing_conclusions']
    list_sentences_citing_discussion = row['sentence_citing_discussion']
    list_sentences_citing_intro = row['sentence_citing_intro']
    list_sentences_citing_maintext = row['sentence_citing_maintext']
    
    if list_sentences_citing_conclusions != None:
        sentences_conclusions = " ".join(list_sentences_citing_conclusions)
        sentences_conclusions = remove_html_tags(sentences_conclusions)
        sentences_conclusions = remove_citations(sentences_conclusions)
    else:
        sentences_conclusions = ""
        
    if list_sentences_citing_discussion != None:
        sentences_discussion = " ".join(list_sentences_citing_discussion)
        sentences_discussion = remove_html_tags(sentences_discussion)
        sentences_discussion = remove_citations(sentences_discussion)

    else:
        sentences_discussion = ""
    
    if list_sentences_citing_intro != None:
        sentences_intro = " ".join(list_sentences_citing_intro)
        sentences_intro = remove_html_tags(sentences_intro)
        sentences_intro = remove_citations(sentences_intro)

    else:
        sentences_intro = ""

    if list_sentences_citing_maintext != None:
        sentences_maintext = " ".join(list_sentences_citing_maintext)
        sentences_maintext = remove_html_tags(sentences_maintext)
        sentences_maintext = remove_citations(sentences_maintext)
    else:
        sentences_maintext = ""

    all_sentences = (sentences_maintext + " " + 
                     sentences_discussion + " " +
                     sentences_intro + " " +
                     sentences_maintext)
    if all_sentences != "   ":
        list_all_citing_citing_sentences.append(all_sentences)
    else:
        list_all_citing_citing_sentences.append(np.nan)

#print(all_sentences)

df_sentences_classification = pd.concat([df_Articles_before_after_retraction[["before_retraction",
                                                                             "after_retraction"]],
                                        pd.DataFrame({"sentences_citing" : list_all_citing_citing_sentences
                                                      })
                                        ], axis = 1)
df_sentences_classification

before_retraction  after_retraction  \
0                    0                 1   
1                    1                 0   
2                    0                 1   
3                    1                 0   
4                    1                 0   
5                    0                 1   
6                    1                 0   
7                    0                 1   
8                    1                 0   
9                    1                 0   
10                   1                 0   
11                   1                 0   
12                   1                 0   
13                   0                 1   
14                   0                 1   
15                   0                 1   
16                   1                 0   
17                   1                 0   
18                   0                 1   
19                   1                 0   
20                   1                 0   
21                   1                 0   
22                   1                 0   
23                   1                 0   
24                   1                 0   
25                   1                 0   
26                   1                 0   
27                   1                 0   
28                   1                 0   
29                   1                 0   
..                 ...               ...   
122                  1                 0   
123                  1                 0   
124                  0                 1   
125                  1                 0   
126                  1                 0   
127                  1                 0   
128                  0                 1   
129                  0                 1   
130                  1                 0   
131                  0                 1   
132                  0                 1   
133                  1                 0   
134                  1                 0   
135                  1                 0   
136                  1                 0   
137                  1                 0   
138                  1                 0   
139                  0                 1   
140                  1                 0   
141                  0                 1   
142                  1                 0   
143                  0                 1   
144                  0                 1   
145                  0                 1   
146                  0                 1   
147                  1                 0   
148                  0                 1   
149                  0                 1   
150                  1                 0   
151                  0                 1   

                                      sentences_citing  
0     1  –  3)  In a survey of 1,397 Korean people,...  
1    To verify the activation of FOXO1 in our model...  
2    Boiling or chlorinating water at the point of ...  
3    in a S961-induced insulin resistance mouse mod...  
4                                                  NaN  
5     Betatrophin, also known as lipasin (  19  ) o...  
6     1  The official name of the gene encoding mou...  
7    A previously reported effect of betatrophin on...  
8     OTHER FACTORS  Betatrophin has been shown to ...  
9     Another example is the Akt signaling cascade ...  
10                                                 NaN  
11    Although the exact mechanism, by which PTEN4A...  
12                                                 NaN  
13    \n     34)\n    \n    Results from our previo...  
14   One quite promising study was published recent...  
15   Additionally, studies conducted by Xiang et al...  
16     created a strain of  B. subtilis  able to gr...  
17   Other recent efforts have described the creati...  
18   One of the intriguing problems discussed inclu...  
19   Others have shown that incongruence with taxon...  
20                                                 NaN  
21                    

In [69]:
#df.dropna(subset = "sentences_citing")

df_sentences_classification = df_sentences_classification.dropna(subset = ["sentences_citing"])

In [70]:
df_sentences_classification.shape

(127, 3)

In [71]:
df_sentences_classification


before_retraction  after_retraction  \
0                    0                 1   
1                    1                 0   
2                    0                 1   
3                    1                 0   
5                    0                 1   
6                    1                 0   
7                    0                 1   
8                    1                 0   
9                    1                 0   
11                   1                 0   
13                   0                 1   
14                   0                 1   
15                   0                 1   
16                   1                 0   
17                   1                 0   
18                   0                 1   
19                   1                 0   
22                   1                 0   
23                   1                 0   
24                   1                 0   
25                   1                 0   
26                   1                 0   
27                   1                 0   
28                   1                 0   
29                   1                 0   
30                   0                 1   
31                   1                 0   
32                   1                 0   
33                   1                 0   
34                   0                 1   
..                 ...               ...   
114                  1                 0   
115                  0                 1   
116                  0                 1   
117                  1                 0   
118                  0                 1   
119                  0                 1   
120                  1                 0   
122                  1                 0   
124                  0                 1   
125                  1                 0   
126                  1                 0   
127                  1                 0   
128                  0                 1   
129                  0                 1   
130                  1                 0   
131                  0                 1   
132                  0                 1   
133                  1                 0   
134                  1                 0   
135                  1                 0   
136                  1                 0   
137                  1                 0   
139                  0                 1   
140                  1                 0   
141                  0                 1   
142                  1                 0   
144                  0                 1   
146                  0                 1   
148                  0                 1   
151                  0                 1   

                                      sentences_citing  
0     1  –  3)  In a survey of 1,397 Korean people,...  
1    To verify the activation of FOXO1 in our model...  
2    Boiling or chlorinating water at the point of ...  
3    in a S961-induced insulin resistance mouse mod...  
5     Betatrophin, also known as lipasin (  19  ) o...  
6     1  The official name of the gene encoding mou...  
7    A previously reported effect of betatrophin on...  
8     OTHER FACTORS  Betatrophin has been shown to ...  
9     Another example is the Akt signaling cascade ...  
11    Although the exact mechanism, by which PTEN4A...  
13    \n     34)\n    \n    Results from our previo...  
14   One quite promising study was published recent...  
15   Additionally, studies conducted by Xiang et al...  
16     created a strain of  B. subtilis  able to gr...  
17   Other recent efforts have described the creati...  
18   One of the intriguing problems discussed inclu...  
19   Others have shown that incongruence with taxon...  
22   RhoA has been reported to be one of the target...  
23   The abnormal expression profiles of miRNAs hav...  
24    Recently, miRNAs have been shown to play a ro...  
25   More and more evidence shows that miR-31 has d...  
26      H 3 K 9 A c   

# Preprocessing data

In [ ]:
for df_sentences_classification